lib/fluent/plugin/elasticsearch_error_handler.rb

```diff
@@ -60,6 +60,8 @@ def handle_error(response, tag, chunk, bulk_message_count, extracted_values)
         write_operation = @plugin.write_operation
       elsif INDEX_OP == @plugin.write_operation && item.is_a?(Hash) && item.has_key?(CREATE_OP)
         write_operation = CREATE_OP
+      elsif UPSERT_OP == @plugin.write_operation && item.is_a?(Hash) && item.has_key?(UPDATE_OP)
```

Comment:
The patch seems to be good.
Just my curiosity, is this modification whether dangerous or safe?

Reply:
I think remarkable case is mainly conflict.
And, I think the main point is whether eventual consistency should be supported or not for all operation.
This is because ElasticsearchErrorHandler seems to be created for supporting eventual consistency to some extent.

If someone think just only upsert should not have eventual consistency, this patch might be harmful.

However, I could not come up with such a case.
So, I think this patch is safe.